mne/decoding/transformer.py

```diff
@@ -322,28 +332,19 @@ def transform(self, epochs_data, y=None):
         Returns
         -------
         psd : array, shape (n_signals, len(freqs)) or (len(freqs),)
-            The computed PSD.
+            The computed PSD. This also creates the attribute `freqs_`, which
+            contains the frequencies in the PSD estimate.
         """
 
         if not isinstance(epochs_data, np.ndarray):
             raise ValueError("epochs_data should be of type ndarray (got %s)."
                              % type(epochs_data))
-
-        epochs_data = np.atleast_3d(epochs_data)
-
-        n_epochs, n_channels, n_times = epochs_data.shape
-        X = epochs_data.reshape(n_epochs * n_channels, n_times)
-
-        psd, _ = multitaper_psd(x=X, sfreq=self.sfreq, fmin=self.fmin,
-                                fmax=self.fmax, bandwidth=self.bandwidth,
-                                adaptive=self.adaptive, low_bias=self.low_bias,
-                                n_jobs=self.n_jobs,
-                                normalization=self.normalization,
-                                verbose=self.verbose)
-
-        _, n_freqs = psd.shape
-        psd = psd.reshape(n_epochs, n_channels, n_freqs)
-
+        psd, freqs = _psd_multitaper(
+            epochs_data, sfreq=self.sfreq, fmin=self.fmin, fmax=self.fmax,
+            bandwidth=self.bandwidth, adaptive=self.adaptive,
+            low_bias=self.low_bias, normalization=self.normalization,
+            n_jobs=self.n_jobs, verbose=self.verbose)
+        self.freqs_ = freqs
```

Comment:
by the way, do you think multitaper_psd should no longer be imported into time_frequency? And instead we would import psd_multitaper and psd_welch? Either way we may need to address some name confusion, since there is now both a multitaper_psd and a _psd_multitaper, both of which operate on arrays. Basically _psd_multitaper just calls multitaper_psd but with some reshaping at the beginning and end.